### ISSUE TITLE
Concurrent :nodeSetup Tasks Fail

### ISSUE THREAD

**Issue by watchwithmike** [srs/gradle-node-plugin#91]
I'm building a series of Gradle based Node projects on Jenkins.  When two of them run concurrently I end up with a failure while it tries to extract node:

```
Execution failed for task ':nodeSetup'.
> Unable to expand TAR '/opt/jenkins/.gradle/caches/modules-2/files-2.1/org.nodejs/node/0.12.7/a861a6ded50c8cda32ac39c3a54699c767742e6/node-0.12.7-linux-x64.tar.gz'
    The tar might be corrupted or it is compressed in an unexpected way.
    By default the tar tree tries to guess the compression based on the file extension.
    If you need to specify the compression explicitly please refer to the DSL reference.
```

I can get around this by restricting my build queue to a single job but I'm wondering if there's any way to resolve this in the build configuration.

**Comment by watchwithmike** [comment]
Here are relevant sections of my `build.gradle` file.

```Gradle
buildscript {
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
  }
  dependencies {
    classpath "com.moowork.gradle:gradle-gulp-plugin:0.11"
  }
}

apply plugin: 'com.moowork.gulp'

node {
    // Version of node to use.
    version = '0.12.7'

    // Enabled the automatic download. False is the default (for now).
    download = true
}
```

**Comment by srs** [comment]
Yes, I know about this problem. Right now it's unpackng into ``<USER_HOME>/.gradle/nodejs`` directory and I think this is a bad idea since it can end up unpacking it at the same time.

We will either need some locking mechanism or maybe unpacking node/npm in another directory. You can set this directory with ``workDir`` setting. Could you test setting that parameter to the following?

    node {
       ...
       workDir = file("${projectDir}/.gradle/nodejs")
       ...
    }

Report back when you tried it with the result and if this will fix it I will change the default ``workDir`` in the next version.

**Comment by watchwithmike** [comment]
@srs This does resolve the concurrence problem I was having with the plugin.  Thanks for the quick response.  I'll track this and remove the `workDir` setting when this is updated in the next release.

**Comment by srs** [comment]
Great. Then I know what to do :-)
tor. 10. des. 2015 kl. 19.13 skrev Mike Dalrymple <notifications@github.com

**Comment by dcoraboeuf** [comment]
Actually, the same problem occurs if you run the build in `--parallel` mode and that several modules are initialising the node environment.

**Comment by srs** [comment]
Great. Then I can test it.
fre. 11. des. 2015 kl. 23.02 skrev Damien Coraboeuf <
notifications@github.com>:

**Comment by dcoraboeuf** [comment]
Hi,

Is there any progress on this issue?

Thanks,
Damien.

**Comment by srs** [comment]
Will add it into the comming 0.12 version.

**Comment by srs** [comment]
This is now fixed in 0.12-SNAPSHOT.
